src/types/annotator.js

```diff
@@ -0,0 +1,31 @@
+/**
+ * Type definitions for objects passes between the annotator and sidebar
+ */
+
+/** @typedef {import("./api").Target} Target */
+
+/**
+ * @typedef AnnotationData
+ * @prop {string} uri
+ * @prop {string} $tag
+ * @prop {string} [$highlight]
+ * @prop {Target[]} target
+ * @prop {Document} document
+ */
+
+/**
+ * @typedef Document
+ * @prop {string} title
+ * @prop {Object[]} link
+ *   @prop {string} link.rel
+ *   @prop {string} link.type
+ * @prop {Object} dc
+ * @prop {Object} eprints
+ * @prop {Object} facebook
+ * @prop {Object} highwire
+ * @prop {Object} prism
+ * @prop {Object} twitter
```

Comment:
On the annotator side of the application this metadata can be generated in two places:

- `src/annotator/plugin/pdf-metadata.js`
- `src/annotator/plugin/document.js`

It looks to me like only the `title` and `link` top-level properties are common between them and therefore the other fields may not be present. From a quick look over those two files it looks like in web pages there will also be `dc`, `eprints`, `facebook`, `highwire`, `prism` and `twitter`. In PDFs there will also be `documentFingerprint`.

Looking at the tests in `src/annotator/plugin/test/document-test.js`, the types of the dc, eprints etc. fields is a map between metadata field names and arrays of values, which can be expressed here as `Object.<string, string[]>`.